scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -234,6 +267,7 @@ public int schedule() {
                             //get back unused nodes to the RManager
                             if (!nodeSet.isEmpty()) {
                                 releaseNodes(currentJob, nodeSet);
+                                freeResourcesNb += (nodeSet.size() + (nodeSet.getExtraNodes() != null ? nodeSet.getExtraNodes().size() : 0));
```

Comment:
Looks like this code is repeated 3 times and it could be wrapped in a private method